src/ZODB/Connection.py

```diff
@@ -559,10 +497,6 @@ def _commit(self, transaction):
             elif oid in self._added:
                 assert obj._p_serial == z64
             elif obj._p_changed:
-                if oid in self._invalidated:
```

Comment:
This and line 603 are just what we talked about: not pre-emptively raising a ConflictError on invalidations, instead letting the normal conflict resolution process take care of that (or not), right?

I can see why this looks like an optimization, but unless it somehow turns out to destroy some workloads I still agree that the simplification is worthwhile.